src/libraries/System.Private.CoreLib/src/System/ValueTuple.cs

```diff
@@ -693,27 +681,23 @@ public bool Equals(ValueTuple<T1, T2, T3> other)
                 && EqualityComparer<T3>.Default.Equals(Item3, other.Item3);
         }
 
-        bool IStructuralEquatable.Equals(object? other, IEqualityComparer comparer)
-        {
-            if (other == null || !(other is ValueTuple<T1, T2, T3>)) return false;
-
-            var objTuple = (ValueTuple<T1, T2, T3>)other;
-
-            return comparer.Equals(Item1, objTuple.Item1)
-                && comparer.Equals(Item2, objTuple.Item2)
-                && comparer.Equals(Item3, objTuple.Item3);
-        }
+        bool IStructuralEquatable.Equals(object? other, IEqualityComparer comparer) =>
+            other is ValueTuple<T1, T2, T3> vt &&
+            comparer.Equals(Item1, vt.Item1) &&
+            comparer.Equals(Item2, vt.Item2) &&
+            comparer.Equals(Item3, vt.Item3);
 
         int IComparable.CompareTo(object? other)
         {
-            if (other == null) return 1;
+            if (other is null) return 1;
 
-            if (!(other is ValueTuple<T1, T2, T3>))
+            if (other is not ValueTuple<T1, T2, T3> objTuple)
             {
                 ThrowHelper.ThrowArgumentException_TupleIncorrectType(this);
+                return 1; // work around CS0170: Use of possibly unassigned field
             }
 
-            return CompareTo((ValueTuple<T1, T2, T3>)other);
+            return CompareTo(objTuple);
         }
```

Comment:
How about structuring these CompareTo methods instead as:
```C#
int IComparable.CompareTo(object? other)
{
    if (other is not null)
    {
        if (other is ValueTuple<T1, T2, T3> vt)
        {
            return CompareTo(vt);
        }

        ThrowHelper.ThrowArgumentException_TupleIncorrectType(this);
    }

    return 1;
}
```
?

Reply:
That's much better, updated.